app/src/main/java/com/sdp/movemeet/Activity/ActivityDescriptionActivityUnregister.java

```diff
@@ -0,0 +1,137 @@
+
+    FirebaseAuth fAuth;
+    private Activity act;
+    private static final String TAG = "ActDescActivity";
+    StorageReference storageReference;
+    ImageView activityImage;
+    String imagePath;
+    ProgressBar progressBar;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_description);
+
+        Intent intent = getIntent();
+
+        if (intent != null) {
+            act = (Activity) intent.getSerializableExtra("activity");
+        }
+
+        createTitleView();
+        createParticipantNumberView();
+        createDescriptionView();
+        createDateView();
+        createAddressView();
+        createSportView();
+        createDurationView();
+        loadActivityHeaderPicture();
+    }
+
+    public void goToLogin(View v) {
+        if (fAuth.getCurrentUser() == null) {
+            startActivity(new Intent(getApplicationContext(), LoginActivity.class)); // sending the user to the "Login" activity
+            finish();
+        }
+    }
+
+    private void createTitleView() {
+        // activityTitle from the activity
+        TextView activityTitle = (TextView) findViewById(R.id.activity_title_description);
+        if (act != null) activityTitle.setText(act.getTitle());
+    }
+
+    private void createParticipantNumberView() {
+        // number of participants from the activity
+        TextView numberParticipantsView = (TextView) findViewById(R.id.activity_number_description);
+        if (act != null) {
+            numberParticipantsView.setText(act.getParticipantId().size() + "/" + act.getNumberParticipant());
+        }
+    }
+
+    private void createDescriptionView() {
+        // description from the activity
+        TextView descriptionView = (TextView) findViewById(R.id.activity_description_description);
+        if (act != null) descriptionView.setText(act.getDescription());
+    }
+
+    private void createDateView() {
+        // date from the activity
+        TextView dateView = (TextView) findViewById(R.id.activity_date_description);
+        if (act != null) {
+            String pattern = "MM/dd/yyyy HH:mm:ss";
+            DateFormat df = new SimpleDateFormat(pattern);
+            String todayAsString = df.format(act.getDate());
+            dateView.setText(todayAsString);
+        }
+    }
+
+    private void createSportView() {
+        TextView sportView = (TextView) findViewById(R.id.activity_sport_description);
+        if (act != null) {
+            sportView.setText(act.getSport().toString());
+        }
+    }
+
+    private void createDurationView() {
+        TextView durationView = (TextView) findViewById(R.id.activity_duration_description);
+        if (act != null) {
+            durationView.setText(String.valueOf((int) act.getDuration()));
+        }
+    }
+
+    private void createAddressView() {
```

Comment:
Do we want to display the date of the activity to unregistered (i.e. unverified and potentially malicious) users? For security reasons, I think we should either avoid displaying the date or the address of the activity, what do you think?

Reply:
Yes, I only remove the organizer name, but we can discuss tonight about it. Jean-Luc create a new layout for this page so I will see it with him.